hera_cal/utils.py

```diff
@@ -1067,10 +1066,36 @@ def red_average(data, reds=None, bl_tol=1.0, wgt_by_int=False, inplace=False,
             DataContainer, summed nsamples
         elif fed a HERAData or UVData:
             HERAData or UVData object, averaged data
+
+    Notes:
+        1. Different polarizations are assumed to be non-redundant.
+        2. Default weighting is nsamples * integration_time * ~flags.
+        3. If wgts Container is fed then they supercede flag and nsample weighting.
     """
     from hera_cal import redcal, datacontainer
+
+    # type checks
+    if not (isinstance(data, datacontainer.DataContainer) or isinstance(data, UVData)):
```

Comment:
You want `issubclass` so HERAData works too: https://docs.python.org/2/library/functions.html#issubclass

Reply:
this also catches `HERAData`, I tried it